src/Rings/mpoly-graded.jl

```diff
@@ -1502,8 +1502,8 @@ mutable struct HilbertData
     @req coefficient_ring(R) isa AbstractAlgebra.Field "The coefficient ring must be a field"
     @req all(is_homogeneous, gens(I)) "The generators of the ideal must be homogeneous"
 
-    G = groebner_assure(I)
-    cf = Singular.hilbert_series_data(G.S, W)
+    S = singular_groebner_generators(I, false, true)
+    cf = Singular.hilbert_series_data(S, W)
```

Comment:
I assume that we want `singular_groebner_generators` here and not just `groebner_assure`. The doc string of `groebner_assure` explains that one should not rely on the basis being defined on the Singular side.
I also assume that we need a global ordering here; otherwise one can change the `true` to `false`.